### PR TITLE
feat: strict TypeScript, CLI hardening, expanded tests, and docs

### PR DESCRIPTION
**Pull request by jimmyandrade**
### Proposed changes

This PR enables stricter TypeScript checking and ships related CLI fixes, test coverage, and documentation updates that surfaced while landing those changes.

**TypeScript**
- Enable `noImplicitAny` and `strictNullChecks` in `tsconfig.json`.
- Fix type errors across CLI, standalone pipeline, glyphs data, and options.
- Type meow CLI flags explicitly via `WebfontCliFlags`.
- Expand `WebfontOptions` with default fields returned by `getOptions`.
- Replace `null` defaults with `undefined` where types expect optional values.
- Add `@types` devDependencies for `xml2js`, `svg2ttf`, `ttf2woff`, `ttf2eot`, and `wawoff2`.
- Type `metadataProvider` as an optional callback (`MetadataProvider`), not `null`.

**CLI fixes and refactor**
- Parse `--formats` / `-f` into a real `Formats` array (JSON or comma-separated) instead of casting a string — fixes incorrect format selection via substring matching.
- Await `fs.promises.writeFile` in the CLI program so writes finish before exit and errors propagate.
- Extract testable CLI logic into `src/cli/program.ts`.
- Correct `--help` short options to `-u, --fontName` and `-f, --formats` (supersedes #613).
- Ignore root `/webfont.svg` in `.gitignore` (CLI artifact when run without `--dest`).

**Tests**
- Grow suite from 67 to **129** tests.
- Add integration tests for all meow short CLI aliases.
- Add unit tests for `program`, `parseFormatsFlag`, `options`, `templateFonts`, and `glyphsData` edge cases.

**Documentation**
- Update README (formats CLI syntax, `metadataProvider`, corrected CLI aliases).
- Add contributor/agent guidelines for user-facing doc updates and PR scope management in `AGENTS.md` and `CONTRIBUTING.md`.

### Related issue

N/A

Supersedes #613.

### Dependencies added/removed (if applicable)

- Add: `@types/xml2js@0.4.14`
- Add: `@types/svg2ttf@5.0.3`
- Add: `@types/ttf2woff@2.0.4`
- Add: `@types/ttf2eot@2.0.2`
- Add: `@types/wawoff2@1.0.2`

---

### Testing

- [x] I have added or updated tests that prove my fix is effective or that my feature works (if applicable)
  - [x] Unit test — 129 Jest tests (CLI integration, program, parseFormatsFlag, standalone modules)
  - [x] Manual test — verified `--help` shows `-u, --fontName` and `-f, --formats`

#### How to test

1. Run `npm test` — build, lint, and all 129 Jest tests should pass.
2. Confirm `tsc --declaration` completes without errors in `postbuild`.
3. Run `node dist/cli.js --help` and confirm `-u, --fontName` and `-f, --formats`.
4. Run `node dist/cli.js src/fixtures/svg-icons -d /tmp/webfont-test -f '["woff2"]'` and confirm only `.woff2` and `.hash` are written (not `.woff`).

#### Test configuration

N/A

---

## Checklist

- [ ] I have added corresponding labels to this PR (like bug, enhancement...);
- [x] My commits follow the [Conventional Commits 1.0 Guidelines](https://www.conventionalcommits.org/en/v1.0.0/);
- [x] My code follows the style guidelines of this project;
- [x] I have performed a self-review of my own code;
- [x] I have mapped technical debts found on my changes;
- [x] I have made changes to the documentation (if applicable);
- [x] My changes generate no new warnings or errors;

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ if (cli.flags.fontName) {
 }
 
 if (cli.flags.formats) {
-  optionsBase.formats = cli.flags.formats as Formats;
+  optionsBase.formats = cli.flags.formats as unknown as Formats;
 }
```
[the remark this answers is not in the document]
Comment (jimmyandrade) [reply src/cli/index.ts]: Good catch — the unsafe cast was a real bug: a JSON string like `"[\"woff2\"]"` behaved like a string at runtime, so `formats.includes("woff")` matched via substring and produced an extra `.woff` file.

Fixed in `parseFormatsFlag` (`src/cli/parseFormatsFlag.ts`), used from `buildOptionsBase` in `src/cli/program.ts`. The flag is now parsed into a real `Formats` array (JSON array or comma-separated values), with validation for known formats.

Tests added:
- `parseFormatsFlag.test.ts` — JSON/comma parsing and invalid format errors
- `program.test.ts` — `buildOptionsBase` passes a parsed array
- `index.test.ts` short `-f` integration test — only `woff2` (+ hash) is written, not `woff`

File: src/cli/index.ts [hunk 2]
```diff
@@ -194,21 +209,29 @@ Promise.resolve()
...
+                file = path.resolve(destTemplate ?? dest);
+              } else if (type === "hash") {
+                file = path.resolve(path.join(dest, `${fontName}.hash`));
+              } else {
+                file = path.resolve(path.join(dest, `${fontName}.${type}`));
+              }
+
+              return fs.writeFile(file, content, () => {
+                Function.prototype();
+              });
```
[the remark this answers is not in the document]
Comment (jimmyandrade) [reply src/cli/index.ts]: Agreed. The callback-based `fs.writeFile` inside `Promise.all` did not wait for completion.

CLI file-writing logic now lives in `writeResultFiles` (`src/cli/program.ts`), which uses `await fs.promises.writeFile` inside `await Promise.all(...)`. Destination creation uses `ensureDestExists` with `fs.promises.access` / `mkdir` as well. `runCli` is fully async and `startCli` awaits it via `.catch()` before exiting.

Tests added in `program.test.ts`:
- `ensureDestExists` — creates dir when `destCreate` is true, throws when missing
- `writeResultFiles` — writes expected files and propagates write errors (e.g. "disk full")

File: src/types/WebfontOptions.ts [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
+  fontName: string;
+  fontStyle: string;
+  fontWeight: string;
+  formats: Formats;
   formatsOptions: FormatsOptions;
+  glyphTransformFn?: GlyphTransformFn;
+  ligatures: boolean;
   maxConcurrency: number;
+  metadata: unknown;
   metadataProvider: null;
```
[the remark this answers is not in the document]
Comment (jimmyandrade) [reply src/types/WebfontOptions.ts]: Correct — `metadataProvider: null` in defaults was inconsistent with `getGlyphsData`, which treats it as an optional callable (`options.metadataProvider || getMetadataService(...)`).

Changes:
- New `MetadataProvider` type in `src/types/MetadataProvider.ts`
- `WebfontOptions.metadataProvider?: MetadataProvider` (optional, not `null`)
- Removed `metadataProvider: null` from `getOptions` defaults in `src/standalone/options.ts`

Tests added:
- `options.test.ts` — custom provider is preserved
- `glyphsData.test.ts` — custom provider is invoked instead of the default metadata service

**Comment by jimmyandrade** [comment]
Supersedes #613 — CLI help now shows the correct short options (`-u, --fontName` and `-f, --formats`), matching the meow flag definitions. README and test mocks were updated as well.
